api-reference/_hidden/dxDataGridColumn/calculateGroupValue.md

```diff
@@ -4,16 +4,170 @@ type: String | function(rowData)
 ---
 ---
 ##### shortDescription
-Specifies a field name or a function that returns a field name or a value to be used for grouping column cells.
+A field name or a function that returns a field name or value by which to group cells. This value is also used to sort **DataGrid**'s groups.
 
 ##### param(rowData): Object
 The current row's data.
 
 ##### return: any
-The calculated value to be used for grouping.
+The calculated value to be used for grouping and group sorting.
+
+---
+
+The **DataGrid** groups its columns by the grouping column. If you want to use custom grouping, change values of this column in the **calculateGroupValue** function. As a result, group cells' headers will display the changed values that may include extra information. To remove it, specify a custom template in the [groupCellTemplate](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#groupCellTemplate) option ([groupCellRender](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#groupCellRender) in React). The example below uses this concept to display the *"California"* group always on top regardless of the sort order:
```

Comment:
I'm not sure why you add "As a result, group cells' headers will display the changed values that may include extra information". Can you explain?

Reply:
Well, this is my suggestion so I'll try to make it a bit more clear. When users implement the calculateGroupValue, the returned value(that we will later pass to group cells headers) may contain information that was not intended to be in the headers. 
In the next sentence, we show a way to avoid this.